### PR TITLE
adding xs

### PR DESCRIPTION
**Pull request by henrydingliu**
## Summary of Changes  
adding xs to enable more flexible slicing, particular on index terms beyond the first

## Related GitHub Issue(s)  
requested by #720 

## Additional Context for Reviewers  

- [x] I passed tests locally for both code (`pytest`) and documentation changes (`uv run jb build docs --builder=custom --custom-builder=doctest`)

---

> [!NOTE]
> **Low Risk**
> Low risk because the change is additive and covered by new tests, but it touches index-slicing behavior where edge cases (full index selection and `drop_level`) could affect downstream expectations.
> 
> **Overview**
> Adds a new `TrianglePandas.xs()` method that mirrors pandas `xs` for **row-index cross-section slicing**, including `level` selection and `drop_level` handling for MultiIndex-style triangle indexes.
> 
> Extends the test suite with `test_xs` to validate behavior for first-level slicing, deeper-level slicing (e.g., `LOB`), and the special case where all index levels are selected (defaulting the index to `Total` when `drop_level=True`).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 632571cb3018c7513a2faca3b9dbdf3a0c0237a7. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by kennethshsu** [comment]
Do you want this in the latest release? I am always playing catch up and can never get the release out...

**Comment by henrydingliu** [comment]
> Do you want this in the latest release? I am always playing catch up and can never get the release out...

we can hold off. i wanna make sure to give gene sometime to see if it works as he intended.

**Comment by genedan** [comment]
Review notes:

1. Could you point out any issues you had with using `drop_level=True` as the default argument? I played around with it and it did what I expected it to do (which was to drop the level), so it seemed fine with me.

2. Add type `str` as an acceptable data type to the `level` parameter e.g., `clrd.xs('comauto', level='GRNAME')`  would do the same thing as `clrd.xs('comauto', level=1)`. This would mimic the behavior of the [2nd to last example](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.xs.html) in pandas. Bonus points if you can get `level` to accept `list` or `tuple`.

3. Add an `axis` argument. This would mostly be used to cross-section the columns, e.g., `clrd.xs('IncurLoss', axis=1)` However, this gets tricky because trying to use `xs` on more than one column doesn't actually work in the pandas analogue:

```python
d = {
    "num_legs": [4, 4, 2, 2],
    "num_wings": [0, 0, 2, 2],
    "num_thumbs": [4, 5, 6, 7],
    "class": ["mammal", "mammal", "mammal", "bird"],
    "animal": ["cat", "dog", "bat", "penguin"],
    "locomotion": ["walks", "walks", "flies", "walks"],
}
df = pd.DataFrame(data=d)
df = df.set_index(["class", "animal", "locomotion"])
df
```
```python
                           num_legs  num_wings  num_thumbs
class  animal  locomotion                                 
mammal cat     walks              4          0           4
       dog     walks              4          0           5
       bat     flies              2          2           6
bird   penguin walks              2          2           7
```

```python
df.xs('num_legs', axis=1)
```

```python
class   animal   locomotion
mammal  cat      walks         4
        dog      walks         4
        bat      flies         2
bird    penguin  walks         2
Name: num_legs, dtype: int64
```

```python
df.xs(('num_legs', 'num_wings'), axis=1)
```

```python
...
KeyError: ('num_legs', 'num_wings')
```

The pandas behavior is undesirable to me because this seems like a feature that should work, and having `clrd.xs(['IncrLoss', 'CumPaidLoss'], axis=1])` would be useful. This is one of those cases where I think it would be ok to not follow pandas to the letter. I don't think `axis=2` or `axis=3` are necessary at the moment.

In summary, the PR works, the rest of the notes are what I'd consider enhancements to what's already been done. @henrydingliu let me know how many of these items you want to do. I'll move any remaining items to another issue if you want to merge and move on to other things.

**Comment by henrydingliu** [comment]
Thanks for the thorough review. 

1. I didn't run into any issues with defaulting `drop_level` to `True`. My [observation](https://github.com/casact/chainladder-python/issues/720#issuecomment-4415870112) was just that the current `triangle.loc` doesn't drop level more often than it drops level. no issues with setting the default to true. 
2. i would count that as a core functionality. so let me look into it. will ping after i get something working. 
3. i left out `axis = 1` because `triangle[['column1_name','column2_name']]` already works. in pandas you can have layers of columns so `df['column_name']` doesn't always work. but we don't have that problem in `triangle`. do you have any examples where we would need `triangle.xs('column_name', axis=1)` over `triangle['column_name']`?

**Comment by henrydingliu** [comment]
@genedan didn't have to look into much. turns out the underlying get_local_level function already took care of it.

File: chainladder/core/pandas.py [hunk 1]
```diff
@@ -445,6 +448,32 @@ def sqrt(self):
...
+        # create the tuple of the indexer
+        _indexer = [slice(None)] * 2
+        _indexer[0] = loc
+        indexer = tuple(_indexer)
+        result = self.iloc[indexer]
+        if new_ax is not None:
+            new_ax_df = new_ax.to_frame(index=None)[new_ax.names]
+            result.index = new_ax_df
+        else:
+            result.index = pd.DataFrame(data=['Total'],columns=['Total'])
```
[the remark this answers is not in the document]
Comment (henrydingliu) [reply chainladder/core/pandas.py:475]: this is hallucination. `get_loc_level` would not return a None ax if there are duplicates in the index

```python
clrd = cl.load_sample('clrd')
original = clrd.index
print(pd.MultiIndex.from_frame(original).get_loc_level(('Agway Ins Co','comauto'), drop_level=True)[1])
new = original.copy()
new.loc[:,'LOB'] = 'comauto'
print(pd.MultiIndex.from_frame(new).get_loc_level(('Agway Ins Co','comauto'), drop_level=True)[1])
```
```python
None
MultiIndex([('Agway Ins Co', 'comauto'),
            ('Agway Ins Co', 'comauto'),
            ('Agway Ins Co', 'comauto'),
            ('Agway Ins Co', 'comauto')],
           names=['GRNAME', 'LOB'])
```

**Comment by genedan** [comment]
The `axis` thing was just the completest streak in me. I was mostly concerned that a user coming from pandas might be frustrated if differences with our library are too great. But as shown before, using `xs` with `axis=1` isn't very useful in pandas anyway so I doubt we'll get a lot of complaints if we don't make use of the argument. So I think we're good to go.

**Review by genedan — APPROVED** [review]
